scootapi/server/serve.go

```diff
@@ -20,16 +23,70 @@ func Serve(handler scoot.Proc, addr string, transportFactory thrift.TTransportFa
 	return server.Serve()
 }
 
-type Handler struct{}
+type Handler struct {
+	queue queue.Queue
+}
 
-func NewHandler() scoot.Proc {
-	return &Handler{}
+func NewHandler(queue queue.Queue) scoot.Proc {
+	return &Handler{queue}
 }
 
-func (h *Handler) RunJob(def *scoot.JobDefinition) (*scoot.Job, error) {
+func (h *Handler) RunJob(def *scoot.JobDefinition) (*scoot.JobId, error) {
```

Comment:
Test cases please.  Using a Mock of the Queue interface makes testing the error cases really easy.